parser-typechecker/src/Unison/Syntax/FileParser.hs

```diff
@@ -69,58 +75,38 @@ file = do
     let (terms, watches) = (reverse termsr, reverse watchesr)
+        -- All locally declared term variables, running example:
+        --   [foo.alice, bar.alice, zonk.bob]
+        fqLocalTerms :: [v]
+        fqLocalTerms = (stanzas0 >>= getVars) <> (view _1 <$> join accessors) 
     -- suffixified local term bindings shadow any same-named thing from the outer codebase scope
     -- example: `foo.bar` in local file scope will shadow `foo.bar` and `bar` in codebase scope
     let (curNames, resolveLocals) =
           ( Names.shadowTerms locals (NamesWithHistory.currentNames names),
             resolveLocals
           )
           where
-            -- All locally declared term variables, running example:
-            --   [foo.alice, bar.alice, zonk.bob]
-            locals0 :: [v]
-            locals0 = stanzas0 >>= getVars
-            -- Groups variables by their suffixes:
-            --   [ (foo.alice, [foo.alice]),
-            --     (bar.alice, [bar.alice])
-            --     (alice, [foo.alice, bar.alice]),
-            --     (zonk.bob, [zonk.bob]),
-            --     (bob, [zonk.bob]) ]
-            varsBySuffix :: Map Name.Name [v]
-            varsBySuffix = List.multimap [(n, v) | v <- locals0, n <- Name.suffixes (Name.unsafeFromVar v)]
-            -- Any unique suffix maps to the corresponding variable. Above, `alice` is not a unique
-            -- suffix, but `bob` is. `foo.alice` and `bob.alice` are both unique suffixes but
-            -- they map to themselves, so we ignore them. In our example, we'll just be left with
-            --   [(bob, Term.var() zonk.bob)]
-            replacements =
-              [ (Name.toVar n, Term.var () v')
-                | (n, nubOrd -> [v']) <- Map.toList varsBySuffix,
-                  Name.toVar n /= v'
-              ]
-            locals = Map.keys varsBySuffix
-            -- This will perform the actual variable replacements for suffixes
-            -- that uniquely identify definitions in the file. It will avoid
-            -- variable capture and respect local shadowing. For example, inside
-            -- `bob -> bob * 42`, `bob` will correctly refer to the lambda parameter.
-            -- and not the `zonk.bob` declared in the file.
+            -- Each unique suffix mapped to its fully qualified name
+            canonicalVars :: Map v v
+            canonicalVars = UFN.variableCanonicalizer fqLocalTerms
+            
+            -- All unique local term name suffixes - these we want to
+            -- avoid resolving to a term that's in the codebase
+            locals :: [Name.Name]
+            locals = (Name.unsafeFromVar <$> Map.keys canonicalVars)
+            
+            -- A function to replace unique local term suffixes with their
+            -- fully qualified name
+            replacements = [ (v, Term.var () v2) | (v,v2) <- Map.toList canonicalVars, v /= v2 ]
```

Comment:
This code was greatly simplified, calling out to `UFN.variableCanonicalizer` for the heavy lifting.